cmake/CMakeLists.txt

```diff
@@ -831,10 +831,9 @@ function(onnxruntime_set_compile_flags target_name)
       # Enable warning
       target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options -Wall>" "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:-Wall>")
       target_compile_options(${target_name} PRIVATE "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:-Wextra>")
-      if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
-        #external/protobuf/src/google/protobuf/arena.h:445:18: error: unused parameter 'p'
-        target_compile_options(${target_name} PRIVATE "-Wno-unused-parameter")
-      endif()
+      # external/protobuf/src/google/protobuf/arena.h:445:18: error: unused parameter 'p'
+      # onnxruntime_src/cmake/external/protobuf/src/google/protobuf/repeated_ptr_field.h:752:66: error: unused parameter ‘prototype’ [-Werror=unused-parameter]
+      target_compile_options(${target_name} PRIVATE "-Wno-unused-parameter")
```

Comment:
I think it is better to only disable the warning for protobuf related code. If you add the compile flag here, our code will start to have such warnings(though they are suppressed).

Reply:
There're 2 types of errors caused by this `_unused-parameter` . More exception info is in the description of this PR.
1. compiling protobuf source code. The scope could be easily narrowed by updating the patch or other solution.
2. source code which include `onnx/onnx_pb.h`.
     So far, there're 44 c++ and 4 Objective-c files include the header, I'm a little afraid that it would increase a lot of maintenance cost.

Reply:
I've patched ONNX directly.